### DISCUSSION TITLE
Dropwizard 2.x and Jetty 10.x

### DISCUSSION THREAD

**Discussion by HappyHummingbird** [dropwizard/dropwizard#5532] [Q&A] [answered]
Hello,

I am currently using Dropwizard `2.0.29` in my project, and I have bumped the jetty libraries to `9.4.47.v20220610` due to a CVE ( [CVE-2022-2047](https://nvd.nist.gov/vuln/detail/CVE-2022-2047) ).
I noticed that Jetty `9.x` is reaching end of community support (https://github.com/eclipse/jetty.project/issues/7958) so I'm wondering if Dropwizard `2.0.x` would work with Jetty `10.x` ?

I had a go a bumping Jetty to 10 in my project, and it doesn't seem to work ( see below ). But I just wanted to double-check here in case I missed something. 

```
java.lang.NoSuchMethodError: io/dropwizard/jetty/MutableServletContextHandler.addLifeCycleListener(Lorg/eclipse/jetty/util/component/LifeCycle$Listener;)V (loaded from file:*** by jdk.internal.loader.ClassLoaders$AppClassLoader@ef533579) called from class io.dropwizard.setup.AdminEnvironment (loaded from file:*** by jdk.internal.loader.ClassLoaders$AppClassLoader@ef533579).
        at io.dropwizard.setup.AdminEnvironment.<init>(AdminEnvironment.java:45)
        at io.dropwizard.setup.Environment.<init>(Environment.java:82)
        at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:52)
        at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:98)
        at io.dropwizard.cli.Cli.run(Cli.java:78)
        at io.dropwizard.Application.run(Application.java:94)
```

Alternatively, should I upgrade to Dropwizard `2.1.x` first and then bump up Jetty ?

Thanks,
HH

**Comment by zUniQueX** [comment, marked as answer]
Hi @HappyHummingbird. Upgrading Jetty from 9.x to 10.x needs some migration steps instead of just upgrading the dependencies.

For example, the `addLifeCycleListener` method was replaced by the `addEventListener` method.

Since upgrading Jetty would introduce incompatible API changes, we plan to release Dropwizard 3.x mainly for the Jetty 10.x upgrade. If you want to use Jetty 10.x in your application, you can check out the latest beta version of Dropwizard 3.0.0.

**Reply by HappyHummingbird** [reply]
Thank you very much for the speedy reply and confirmation @zUniQueX ! I'll look into Dropwizard `3.0.0`.
